src/components/Contacts/ContactDetails/ContactDontationsTab/DonationsGraph/DonationsGraph.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable eqeqeq */
+import { Box, styled, Typography } from '@material-ui/core';
+import { DateTime } from 'luxon';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Bar,
+  BarChart,
+  CartesianGrid,
+  Legend,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import theme from '../../../../../theme';
+import { DonationsContactFragment } from '../ContactDonationsTab.generated';
+
+const LegendText = styled(Typography)(({ theme }) => ({
+  margin: theme.spacing(3, 0),
+  writingMode: 'vertical-rl',
+  textOrientation: 'mixed',
+}));
+
+const GraphContainer = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  margin: theme.spacing(0, 2, 0, 0),
+}));
+
+const NoDonationsBox = styled(Box)(({ theme }) => ({
+  margin: theme.spacing(4),
+}));
+interface DonationsGraphProps {
+  donations: DonationsContactFragment | null;
+}
+
+export const DonationsGraph: React.FC<DonationsGraphProps> = ({
+  donations,
+}) => {
+  const { t } = useTranslation();
+  const donationsForThisYear = donations?.nodes.filter((donation) => {
+    return (
+      DateTime.fromISO(donation.donationDate).toJSDate() >
+      DateTime.local().minus({ year: 1 }).toJSDate()
+    );
+  });
+
+  const donationsForLastYear = donations?.nodes.filter((donation) => {
+    return (
+      DateTime.fromISO(donation.donationDate).toJSDate() <
+        DateTime.local().minus({ year: 1 }).toJSDate() ||
+      DateTime.fromISO(donation.donationDate).toJSDate() >
+        DateTime.local().minus({ year: 2 }).toJSDate()
+    );
+  });
```

Comment:
This can often be something that's worth adding filters for server side. @soberstadt you have any thoughts?

You could do something like
```graphql
  contact(...){
    ...
    donationsForThisYear: donations(interval: '<ISO8601 Time Interval representing this year>') {
      nodes {
        ...ContactDonation
      }
    }
    donationsForLastYear: donations(interval: '<ISO8601 Time Interval representing last year>') {
      nodes {
        ...ContactDonation
      }
    }
  }
```

Also we may run into pagination issues here as only 25 nodes are returned by default and the API will only return a max of 50 nodes at a time if `first: 50` is specified. I haven't really thought about loading chart data with GraphQL before.